src/twisted/names/test/test_names.py

```diff
@@ -469,6 +502,66 @@ def test_zoneTransferConnectionFails(self):
             resolver.lookupZone("impossible.invalid"), error.DNSLookupError
         )
 
+    def _testHelper_zoneTransferFailure(
+        self,
+        protocolClass: Type[AccumulatingProtocol],
+        expectedFailure: Type[Exception],
+        **kwargs: Any,
+    ) -> defer.Deferred:
+        factory = RecordingServerFactory.forProtocol(protocolClass)
+        nonresponsiveTCP = reactor.listenTCP(0, factory, interface="127.0.0.1")  # type: ignore[attr-defined]
```

Comment:
Could this be done with [`MemoryReactor`](https://twistedmatrix.com/documents/current/api/twisted.internet.testing.MemoryReactor.html) instead of the real reactor? There is a version that provides `IReactorUDP` in `twisted.names.test.test_util`.

Reply:
`listenTCP(0, ...)` will fail sometimes with the real reactor which is further reason to use a test double instead.